without this patch, the jpeg_decompress_struct & jpec_compress_struct will be padded, this in turn
results in a mismatch when the jpeg filter in svtools is built, where no
padding is assumed. Only affects Mac/PPC apparenlty, see fdo#47035
--- misc/jpeg-8c/jpeglib.h	2010-11-17 22:01:56.000000000 +0100
+++ misc/build/jpeg-8c/jpeglib.h	2012-05-19 13:58:31.000000000 +0200
@@ -288,9 +288,9 @@
   JDIMENSION image_width;	/* input image width */
   JDIMENSION image_height;	/* input image height */
   int input_components;		/* # of color components in input image */
-  J_COLOR_SPACE in_color_space;	/* colorspace of input image */
 
   double input_gamma;		/* image gamma of input image */
+  J_COLOR_SPACE in_color_space;	/* colorspace of input image */
 
   /* Compression parameters --- these fields must be set before calling
    * jpeg_start_compress().  We recommend calling jpeg_set_defaults() to
@@ -359,16 +359,16 @@
   /* Parameters controlling emission of special markers. */
 
   boolean write_JFIF_header;	/* should a JFIF marker be written? */
+  boolean write_Adobe_marker;	/* should an Adobe marker be written? */
   UINT8 JFIF_major_version;	/* What to write for the JFIF version number */
   UINT8 JFIF_minor_version;
   /* These three values are not used by the JPEG code, merely copied */
   /* into the JFIF APP0 marker.  density_unit can be 0 for unknown, */
   /* 1 for dots/inch, or 2 for dots/cm.  Note that the pixel aspect */
   /* ratio is defined by X_density/Y_density even when density_unit=0. */
-  UINT8 density_unit;		/* JFIF code for pixel size units */
   UINT16 X_density;		/* Horizontal pixel density */
   UINT16 Y_density;		/* Vertical pixel density */
-  boolean write_Adobe_marker;	/* should an Adobe marker be written? */
+  UINT8 density_unit;		/* JFIF code for pixel size units */
   
   /* State variable: index of next scanline to be written to
    * jpeg_write_scanlines().  Application may use this to control its
@@ -583,11 +583,11 @@
   /* Data copied from JFIF marker; only valid if saw_JFIF_marker is TRUE: */
   UINT8 JFIF_major_version;	/* JFIF version number */
   UINT8 JFIF_minor_version;
+  UINT8 Adobe_transform;	/* Color transform code from Adobe marker */
   UINT8 density_unit;		/* JFIF code for pixel size units */
   UINT16 X_density;		/* Horizontal pixel density */
   UINT16 Y_density;		/* Vertical pixel density */
   boolean saw_Adobe_marker;	/* TRUE iff an Adobe APP14 marker was found */
-  UINT8 Adobe_transform;	/* Color transform code from Adobe marker */
 
   boolean CCIR601_sampling;	/* TRUE=first samples are cosited */
 
